pkg/resource/table/hooks.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package table
+
+import (
+	"errors"
+	"time"
+
+	"github.com/aws-controllers-k8s/dynamodb-controller/apis/v1alpha1"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+)
+
+var (
+	ErrTableDeleting = errors.New("table in 'DELETING' state, cannot be modified or deleted")
+	ErrTableCreating = errors.New("table in 'CREATING' state, cannot be modified or deleted")
+	ErrTableUpdating = errors.New("table in 'UPDATING' state, cannot be modified or deleted")
```

Comment:
nit:
```suggestion
	ErrTableDeleting = errors.New("Table in 'DELETING' state, cannot be modified or deleted")
	ErrTableCreating = errors.New("Table in 'CREATING' state, cannot be modified or deleted")
	ErrTableUpdating = errors.New("Table in 'UPDATING' state, cannot be modified or deleted")
```

Reply:
I always thought that Go errors should never be capitalized but actually there is an exception.. proper nouns or acronyms. source:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
My thought too, `Table` is the resource name.